ofronds_internals/ofronds.ml

```diff
@@ -95,6 +96,34 @@ let version =
   let open Build_info.V1 in
   match version () with Some v -> Version.to_string v | None -> "dev"
 
+let hint =
+  let doc =
+    "Get a hint if you're stuck on a problem by writing `ofronds hint \
+     <name_of_exercise>"
+  in
+  let exercise_name =
+    let doc = "Name of the exercise" in
+    (* Arg.required @@ Arg.pos 0 (Arg.some ) None
+       @@ Arg.info ~doc *)
+    Cmdliner.Arg.(value & pos 0 string "hint" & info [] ~doc)
+  in
+  let fn exercise_name =
+    let metadata = Lazy.force exercise_metadata in
+    let hastable = Exercise.Set.to_hashtable metadata in
+    match Hashtbl.find_opt hastable exercise_name with
+    | Some exercise -> (
+        match Exercise.hint exercise with
+        | Some hint -> Fmt.pr "%s\n" hint
+        | None ->
+            Fmt.pr "%s\n"
+              "There's no hint for this exercise, if you think it'd be useful \
+               please open an issue.")
+    | None ->
+        User_message.failf
+          "This exercise doesn't exist, perphaps you entered the name wrong?"
```

Comment:
As mentioned in the general comment, I think the logic of this should be in `Exercise`.